### PR TITLE
Add authorization successful message

### PR DESCRIPTION
**Pull request by joneug**
See #34.

### CODE REVIEW

File: templates/authorization-successful.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+style('oauth2', 'main');
+style('oauth2', 'authorization');
+?>
+
+<div id="app">
+    <div id="app-content">
+        <div id="app-content-wrapper">
+            <div id="authorization-dialog">
+                <p><b><?php p($l->t('The application was authorized successfully. You can now close this site.'));?></b></p>
```
Comment (ogoffart) [inline templates/authorization-successful.php]: Maybe it should be s/site/page/   or "window"  or "tab" ?

Comment (joneug) [inline templates/authorization-successful.php]: Done.

**Comment by ogoffart** [comment]
I have implemented the code that reads this new property in https://github.com/owncloud/client/commit/17d14d1facff65dfadd55305c62f812c6b6310fe
Everything worked as expected.

May I suggest to rename the property `authorization_successful_message_url`  to simply `message_url`,  this way this property can be re-used also if there was an error.
Other than that, I am not qualified to review PHP code.

File: lib/Controller/OAuthApiController.php [hunk 2]
```diff
@@ -49,32 +50,38 @@ class OAuthApiController extends ApiController {
 	/** @var RefreshTokenMapper */
 	private $refreshTokenMapper;
 
+    /** @var IURLGenerator */
+    private $urlGenerator;
```
Comment (DeepDiver1975) [inline lib/Controller/OAuthApiController.php]: indent - please adjust - we use tabs - no spaces - thx

File: lib/Controller/OAuthApiController.php [hunk 3]
```diff
@@ -49,32 +50,38 @@ class OAuthApiController extends ApiController {
...
-	 *
-	 * @param string $AppName The app's name.
-	 * @param IRequest $request The request.
-	 * @param ClientMapper $clientMapper The client mapper.
-	 * @param AuthorizationCodeMapper $authorizationCodeMapper The authorization code mapper.
-	 * @param AccessTokenMapper $accessTokenMapper The access token mapper.
-	 * @param RefreshTokenMapper $refreshTokenMapper The refresh token mapper.
-	 * @param ILogger $logger The logger.
-	 */
+    /**
```
Comment (DeepDiver1975) [inline lib/Controller/OAuthApiController.php]: indent as well - please adjust - thx

**Comment by joneug** [comment]
> - indent is off - please adjust

Done!

**Comment by ogoffart** [comment]
The client commit was updated with the new property name. And I also made a pull request for the client that depends on this one: https://github.com/owncloud/client/pull/5876

**Comment by ogoffart** [comment]
The corresponding merge request was merged in the client already.
Anything  else blocking this to be merged?

**Comment by DeepDiver1975** [comment]
👍

**Comment by guruz** [comment]
@DeepDiver1975 @ogoffart Have you considered if Javascript could be used to close the browser window/tab again? 

Or maybe that's too confusing if the re-popping up of the client wizard did not work.
Or in general to confusing..

OK, ignore that idea

**Comment by ogoffart** [comment]
@guruz  unfortunately, closing a window from javascript is not allowed.
There is `window.close()`,  but:

https://www.w3.org/TR/html51/browsers.html#dom-window-close
> A browsing context is script-closable if it is an auxiliary browsing context that was created by a script (as opposed to by an action of the user), or if it is a top-level browsing context whose session history contains only one Document.

Since we are coming from the authorization page, it is hard to do make it work.

**Pull request merged**